apps/tangle-cloud/src/pages/blueprints/[id]/deploy/page.tsx

```diff
@@ -21,16 +21,27 @@ import { z } from 'zod';
 import useServiceRequestTx from '../../../../data/services/useServicesRequestTx';
 import { TxStatus } from '@tangle-network/tangle-shared-ui/hooks/useSubstrateTx';
 import { PagePath } from '../../../../types';
-import useNetworkStore from '@tangle-network/tangle-shared-ui/context/useNetworkStore';
 import { getApiPromise } from '@tangle-network/tangle-shared-ui/utils/polkadot/api';
+import useNetworkStore from '@tangle-network/tangle-shared-ui/context/useNetworkStore';
 
 const DeployPage: FC = () => {
-  const { id } = useParams();
+  const { id: idParam } = useParams();
   const navigate = useNavigate();
+
   const wsRpcEndpoint = useNetworkStore(
     (store) => store.network2?.wsRpcEndpoint,
   );
 
+  const id = useMemo(() => {
```

Comment:
This is repeated in multiple places, a generic hook could help reduce redundancy.

```tsx
import { useMemo } from "react";
import { useParams } from "react-router-dom";
import { ZodType } from "zod";

export function useParam<T>(
  paramName: string,
  schema: ZodType<T>
): T | undefined {
  const params = useParams();

  return useMemo(() => {
    const paramValue = params[paramName];
    if (paramValue === undefined) return undefined;

    const result = schema.safeParse(paramValue);
    return result.success ? result.data : undefined;
  }, [params, paramName, schema]);
}
```

```tsx
import { z } from "zod";
import { useParam } from "./hooks/useParam";

const id = useParam("id", z.coerce.bigint());
```